### PR TITLE
Added conditions before deleting template elements

### LINKED ISSUE #206: Check for Template elements before deleting them in generate_yaml.py
We need to support the scenario where a user uses their own template yaml files to define their appwrappers.  Due to this scenario, we should not presume the existence of any fields in `generate_yaml.py` and should check for their existence before deleting them. For example,  `disable_raycluster_tls(resources)` will fail if a custom template does not define a volume. 

https://github.com/project-codeflare/codeflare-sdk/blob/ec6c787c6bc060fccb82f9b6e0684d3dfc12e016/src/codeflare_sdk/utils/generate_yaml.py#L263C1-L283

### PR DESCRIPTION
**Pull request by Bobbins228**
closes #206

### CODE REVIEW

**Review by MichaelClifford — CHANGES_REQUESTED** [review]
Looks good, but one issue with the way the last couple checks are done.

File: src/codeflare_sdk/utils/generate_yaml.py [hunk 1]
```diff
@@ -302,14 +302,25 @@ def disable_raycluster_tls(resources):
...
-        del resources["GenericItems"][2]  # ca-secret
+    updated_items = []
+    for i in resources["GenericItems"][:]:
+        if "rayclient-deployment-name" in i["generictemplate"]["metadata"]["name"]:
+            continue
+        if "ca-secret-deployment-name" in i["generictemplate"]["metadata"]["name"]:
+            continue
+        updated_items.append(i)
+
+    resources["GenericItems"] = updated_items
```
Comment (MichaelClifford) [inline src/codeflare_sdk/utils/generate_yaml.py:323]: Why aren't you deleting these items? @Bobbins228

Comment (Bobbins228) [reply src/codeflare_sdk/utils/generate_yaml.py:323]: It's essentially the same right? 
`resources["GenericItems"]` is equated to `resources["GenericItems"]` without the ca-secret and ray-client. 

I tested this out and compared the yaml file with one generated with the current sdk and there were no differences. 

I could do what I did [here](https://github.com/project-codeflare/codeflare-sdk/pull/244/files#diff-279fd895cfcb8295f1b610826f8eb7a40fd543ff873661acd9dc452cef3d9d14R305-R313) if necessary.

**Review by MichaelClifford — APPROVED** [review]
LGTM!

**Pull request merged**